terraform/eval_validate.go

```diff
@@ -101,6 +121,64 @@ func (n *EvalValidateProvisioner) Eval(ctx EvalContext) (interface{}, error) {
 	}
 }
 
+func (n *EvalValidateProvisioner) validateConnConfig(connConfig *ResourceConfig) (warns []string, errs []error) {
```

Comment:
Hmm, that's what I was afraid of. 

`connConfig.Config` is already just a `map[string]interface{}`; couldn't this just be a function like 

    communicator.ValidateConnConfig(cfg map[string]interface{}) error

 and pass it `ResourceConfig.Config`?